### ISSUE TITLE
Unable to build project using VS 17.12.3: “DiscoverPrecompressedAssets” task failed unexpectedly

### ISSUE THREAD

**Issue by fltrierweiler** [dotnet/aspnetcore#59625]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Describe the bug

I've found a [similar issue,](https://github.com/dotnet/aspnetcore/issues/59476) but the exception there isn't the same (`System.ArgumentException` there and `System.InvalidOperationException` here) and that issue was closed without an answer, so I've decided to create a new one.

Basically, I'm unable to build an existing project that uses `Blazor.BrowserExtension.Template`. It used to work fine with version 17.11.35219.272.

### Expected Behavior

Simply put, the project build should work without errors, as it used to do.

### Steps To Reproduce

I've created a [minimalistic project to showcase the issue](https://github.com/fltrierweiler/ExtensionTest). To reproduce the error, all that has to be done is to build the project.

### Exceptions (if any)

Here's the error I get whenever I try to build the project using VS 17.12.3:
```
The "DiscoverPrecompressedAssets" task failed unexpectedly.
System.InvalidOperationException: Fingerprint for '...\ExtensionTest\ExtensionTest\obj\Debug\net8.0\browserextension\build\BackgroundWorker.js' is not defined.
   at Microsoft.AspNetCore.StaticWebAssets.Tasks.StaticWebAsset.Validate()
   at Microsoft.AspNetCore.StaticWebAssets.Tasks.StaticWebAsset.FromTaskItem(ITaskItem item)
   at System.Linq.Enumerable.WhereSelectArrayIterator`2.MoveNext()
   at System.Linq.Buffer`1..ctor(IEnumerable`1 source)
   at System.Linq.Enumerable.ToArray[TSource](IEnumerable`1 source)
   at Microsoft.AspNetCore.StaticWebAssets.Tasks.DiscoverPrecompressedAssets.Execute()
   at Microsoft.Build.BackEnd.TaskExecutionHost.Execute()
   at Microsoft.Build.BackEnd.TaskBuilder.<ExecuteInstantiatedTask>d__26.MoveNext()
```

### .NET Version

9.0.101

### Anything else?

UPDATE: If I use SDK 8.0.404 and specify the version by adding a `global.json` to the solution folder, it actually works. Still no way to work with 9.0.101, though.

**Comment by MarcoMedrano** [comment]
I have the same issue building my maui app, your workaround is working (as I did not update my projects to dotnet 9):

global.json at root directory:
```json
{
  "sdk": {
    "version": "8.0.100"
  }
}
```

**Comment by lofcz** [comment]
Hitting a similar error after upgrading from 8 to 9:

```
Error MSB4018 : The "DiscoverPrecompressedAssets" task failed unexpectedly.
System.ArgumentException: An item with the same key has already been added. Key: [path]\wwwroot\app.css
   at System.Collections.Generic.Dictionary`2.TryInsert(TKey key, TValue value, InsertionBehavior behavior)
   at System.Collections.Generic.Dictionary`2.Add(TKey key, TValue value)
   at System.Linq.Enumerable.SpanToDictionary[TSource,TKey](ReadOnlySpan`1 source, Func`2 keySelector, IEqualityComparer`1 comparer)
   at System.Linq.Enumerable.ToDictionary[TSource,TKey](IEnumerable`1 source, Func`2 keySelector, IEqualityComparer`1 comparer)
   at Microsoft.AspNetCore.StaticWebAssets.Tasks.DiscoverPrecompressedAssets.Execute()
   at Microsoft.Build.BackEnd.TaskExecutionHost.Execute()
   at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(TaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask)
```

In csproj, I have several files linked as:
```
<Content Include="..\[path]\wwwroot\**\app.scss">
    <Link>wwwroot\%(RecursiveDir)%(Filename)%(Extension)</Link>
    <CopyToPublishDirectory>PreserveNewest</CopyToPublishDirectory>
    <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
</Content>
```

If I remove these links, the app builds but I can't use assets from the web project in the sln. This works as expected on 8.0.X

**Comment by evgenyvalavin** [comment]
+1

**Comment by SirBirne** [comment]
This issue has persisted since SDK > 8.0.403 and I'm also looking forward to a fix.

**Comment by Lionel-Renaux** [comment]
As Github Copilot is being removed in 17.11, this is now beginning to be an issue as we cannot update without breaking our builds. The fix is in fact quite simple and already 2 PRs are open to fix it!

**Comment by LeeStevens318** [comment]
bump

**Comment by tbaggett-ebsco** [comment]
> For anybody encountering this issue, as a workaround you can add a target to the `Directory.Build.targets` or the `.csproj` file which removes all duplicates from `StaticWebAsset`:
> 
> 
>     <Target
>             Name="Workaround_ResolveBuildCompressedStaticWebAssetsConfiguration"
>             BeforeTargets="ResolveBuildCompressedStaticWebAssetsConfiguration">
>         
>         
>         <ItemGroup>
>             <StaticWebAsset_NoDuplicates />
>         </ItemGroup>
>         
>         <RemoveDuplicates Inputs="@(StaticWebAsset)">
>             <Output TaskParameter="Filtered" ItemName="StaticWebAsset_NoDuplicates" />
>         </RemoveDuplicates>
>         
>         <ItemGroup>
>             
>             <StaticWebAsset Remove="@(StaticWebAsset)" />
>             
>             
>             <StaticWebAsset Include="@(StaticWebAsset_NoDuplicates)" />
>         </ItemGroup>
>     </Target>

Thanks for this info! I will give it a try

**Comment by tbaggett-ebsco** [comment]
@G-MW, I confirmed your workaround resolved the issue for me. Thanks again for sharing it!

**Comment by iAmBipinPaul** [comment]
I'm getting this after recent VS update  Visual Studio 17.14.0

  _Microsoft.NET.Sdk.StaticWebAssets.Compression.targets(289, 5): [MSB4018] The "DiscoverPrecompressedAssets" task failed unexpectedly.
System.InvalidOperationException: File length for `'.........\obj\Debug\net9.0\RazorSourceMapFiles\b-07r5srnkv9.txt' `is not defined.
   at Microsoft.AspNetCore.StaticWebAssets.Tasks.StaticWebAsset.Validate()
   at Microsoft.AspNetCore.StaticWebAssets.Tasks.StaticWebAsset.FromTaskItem(ITaskItem item)
   at System.Linq.Enumerable.ArraySelectIterator`2.ToArray()
   at System.Linq.Enumerable.ToArray[TSource](IEnumerable`1 source)
   at Microsoft.AspNetCore.StaticWebAssets.Tasks.DiscoverPrecompressedAssets.Execute()
   at Microsoft.Build.BackEnd.TaskExecutionHost.Execute()
   at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(TaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask)_

**Comment by iAmBipinPaul** [comment]
Uninstalling VS and then installing SDK 9.0.204 fixed the issue, so issue I'm facing is with SDK 9.0.300

**Comment by bwinklesky** [comment]
Here to comment on this lovely error I received after patch Tuesday.  Why does every release break everything?

**Comment by lbrpsoftware** [comment]
Same Issue in:

- SDK v9.0.300
- VS 17.14.0:

Removing build paths (bin, obj, .vs) works again for a while.

**Comment by jtsom-do** [comment]
Came to say "Me too".

It does look like this started with the update to 9.0.300.

In my case, I see this in VSCode, running a simple API application. If I hit F5 to start the app, I get the error and the build fails.

If I then go into a terminal and enter the command `dotnet build` - the build will complete without error. Hitting `F5` again will build and launch the app.

However stopping the app, making any edit and trying to `F5` again will show the error again.

I will try the work-around.

FWIW, this simple app has very little in the wwwroot folder (just basic css, etc.)

![Image](https://github.com/user-attachments/assets/e1cbac93-9110-4315-8341-ecfe804e9f91)

**Comment by 1230landon** [comment]
I went to the Visual Studio Insaller, to Individual Components, at the very bottom I changed my Windows 11 SDK (10.0.26100.0) to Windows 11 SDK (10.0.22621.0) (I changed other thigs but I think this is what really got it. I then went back to my project and it didint work until a Ctrl+Shift+B. Runs great now. 
Other changes: Added .NET 6.0 Runtime (Out of Support). 
I Compared this off of a laptop that had a working version. VS 17.13.6 is on there while VS 17.14.0 is on here. So if my changes do not work, I suggest finding a computer with a slightly older VS

**Comment by martincostello** [comment]
Similar error in VS 2022 17.14 Preview 7 and the 9.0.300 SDK, but a different stack trace to the initial description's:

```
The "DiscoverPrecompressedAssets" task failed unexpectedly.
System.ArgumentException: An item with the same key has already been added.
   at System.ThrowHelper.ThrowArgumentException(ExceptionResource resource)
   at System.Collections.Generic.Dictionary`2.Insert(TKey key, TValue value, Boolean add)
   at System.Linq.Enumerable.ToDictionary[TSource,TKey,TElement](IEnumerable`1 source, Func`2 keySelector, Func`2 elementSelector, IEqualityComparer`1 comparer)
   at Microsoft.AspNetCore.StaticWebAssets.Tasks.DiscoverPrecompressedAssets.Execute()
   at Microsoft.Build.BackEnd.TaskExecutionHost.Execute()
   at Microsoft.Build.BackEnd.TaskBuilder.<ExecuteInstantiatedTask>d__26.MoveNext()
```

**Comment by jtsom-do** [comment]
The only way I could consistently work around this bug was, since this is just a test app, I moved the static files I had (in the wwwroot folder - originally I had an HTML file in there in order to deal with .NET removing built-in Swagger ... booo) out of the project, and then I had to remove the `app.UseStaticFiles();` (even with the wwwroot folder removed, even just calling `UseStaticFiles()` was somehow caching the "bad" files and I still got the errors, referencing the now-removed files).

Something got messed up in 9.0.300...

**Comment by marqdouj** [comment]
I just updated to .NET 10 Preview 4 SDK and VS 2022 17.4 Preview 7

I am now getting the same issue with a NuGet package I created.
Is there something I need to do to in the package to prevent this?

**Comment by marqdouj** [comment]
Not sure what the 'magic' changes were, but it all works now.
I started changing different Build properties on the tsconfig.json, package.json, etc. and how typescript builds the *.js and how webpack bundles the output.
Now Visual Studio packages both my scripts as 'staticwebassets' and removes the config/configFiles sections from the package; which makes everything work now. 

My GitHub project is [here](https://github.com/marqdouj/JSLogger)

**Comment by davidfowl** [comment]
Repros for me as well.

**Comment by aminmashayekhi** [comment]
same issue with dotnet build and just one custom css in new asp.net core web app (razor pages):

```
PS D:\source\WebApplication4\WebApplication4> dotnet build
Restore complete (0/3s)
  WebApplication4 failed with 1 error(s) (0/2s)
    C:\Program Files\dotnet\sdk\9.0.300\Sdks\Microsoft.NET.Sdk.StaticWebAssets\targets\Microsoft.NET.Sdk.StaticWebAssets.targets(693,5): error MSB4018:
      The "DefineStaticWebAssetEndpoints" task failed unexpectedly.
      System.AggregateException: One or more errors occurred. (Last write time for 'D:\source\WebApplication4\WebApplica
      tion4\wwwroot\css\request-page.css' is not defined.) (Last write time for 'D:\source\WebApplication4\WebApplicatio
      n4\wwwroot\images\banner3.svg' is not defined.)
       ---> System.InvalidOperationException: Last write time for 'D:\source\WebApplication4\WebApplication4\wwwroot\css
      \request-page.css' is not defined.
         at Microsoft.AspNetCore.StaticWebAssets.Tasks.StaticWebAsset.Validate()
         at Microsoft.AspNetCore.StaticWebAssets.Tasks.StaticWebAsset.FromTaskItem(ITaskItem item)
         at Microsoft.AspNetCore.StaticWebAssets.Tasks.DefineStaticWebAssetEndpoints.ParallelWorker.Process(Int32 i, Par
      allelLoopState _)
         at Microsoft.AspNetCore.StaticWebAssets.Tasks.DefineStaticWebAssetEndpoints.<>c.<Execute>b__16_2(Int32 i, Paral
      lelLoopState loop, ParallelWorker state)
         at System.Threading.Tasks.Parallel.<>c__DisplayClass19_0`2.<ForWorker>b__1(RangeWorker& currentWorker, Int64 ti
      meout, Boolean& replicationDelegateYieldedBeforeCompletion)
      --- End of stack trace from previous location ---
         at System.Threading.Tasks.Parallel.<>c__DisplayClass19_0`2.<ForWorker>b__1(RangeWorker& currentWorker, Int64 ti
      meout, Boolean& replicationDelegateYieldedBeforeCompletion)
         at System.Threading.Tasks.TaskReplicator.Replica.Execute()
         --- End of inner exception stack trace ---
         at System.Threading.Tasks.TaskReplicator.Run[TState](ReplicatableUserAction`1 action, ParallelOptions options,
      Boolean stopOnFirstFailure)
         at System.Threading.Tasks.Parallel.ForWorker[TLocal,TInt](TInt fromInclusive, TInt toExclusive, ParallelOptions
       parallelOptions, Action`1 body, Action`2 bodyWithState, Func`4 bodyWithLocal, Func`1 localInit, Action`1 localFin
      ally)
      --- End of stack trace from previous location ---
         at System.Threading.Tasks.Parallel.ForWorker[TLocal,TInt](TInt fromInclusive, TInt toExclusive, ParallelOptions
       parallelOptions, Action`1 body, Action`2 bodyWithState, Func`4 bodyWithLocal, Func`1 localInit, Action`1 localFin
      ally)
         at System.Threading.Tasks.Parallel.For[TLocal](Int32 fromInclusive, Int32 toExclusive, Func`1 localInit, Func`4
       body, Action`1 localFinally)
         at Microsoft.AspNetCore.StaticWebAssets.Tasks.DefineStaticWebAssetEndpoints.Execute()
         at Microsoft.Build.BackEnd.TaskExecutionHost.Execute()
         at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(TaskExecutionHost taskExecutionHost, TaskLogging
      Context taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask)
       ---> (Inner Exception #1) System.InvalidOperationException: Last write time for 'D:\source\WebApplication4\WebApp
      lication4\wwwroot\images\banner3.svg' is not defined.
         at Microsoft.AspNetCore.StaticWebAssets.Tasks.StaticWebAsset.Validate()
         at Microsoft.AspNetCore.StaticWebAssets.Tasks.StaticWebAsset.FromTaskItem(ITaskItem item)
         at Microsoft.AspNetCore.StaticWebAssets.Tasks.DefineStaticWebAssetEndpoints.ParallelWorker.Process(Int32 i, Par
      allelLoopState _)
         at Microsoft.AspNetCore.StaticWebAssets.Tasks.DefineStaticWebAssetEndpoints.<>c.<Execute>b__16_2(Int32 i, Paral
      lelLoopState loop, ParallelWorker state)
         at System.Threading.Tasks.Parallel.<>c__DisplayClass19_0`2.<ForWorker>b__1(RangeWorker& currentWorker, Int64 ti
      meout, Boolean& replicationDelegateYieldedBeforeCompletion)
      --- End of stack trace from previous location ---
         at System.Threading.Tasks.Parallel.<>c__DisplayClass19_0`2.<ForWorker>b__1(RangeWorker& currentWorker, Int64 ti
      meout, Boolean& replicationDelegateYieldedBeforeCompletion)
         at System.Threading.Tasks.TaskReplicator.Replica.Execute()<---


Build failed with 1 error(s) in 1/2s
```

**Comment by chopra-design** [comment]
bump.

This error is happening consistently on the latest update. Unable to run a project. Temporary workaround is to "Clean Build" then run it and that sometimes works or to restart visual studio if it gets stuck. Nothing works consistently.

Microsoft Visual Studio Community 2022 (64-bit) - Current
Version 17.14.1

Error:
Severity	Code	Description	Project	File	Line	Suppression State
Error (active)	MSB4018	The "DiscoverPrecompressedAssets" task failed unexpectedly.
System.ArgumentException: An item with the same key has already been added.
   at System.ThrowHelper.ThrowArgumentException(ExceptionResource resource)
   at System.Collections.Generic.Dictionary`2.Insert(TKey key, TValue value, Boolean add)
   at System.Linq.Enumerable.ToDictionary[TSource,TKey,TElement](IEnumerable`1 source, Func`2 keySelector, Func`2 elementSelector, IEqualityComparer`1 comparer)
   at Microsoft.AspNetCore.StaticWebAssets.Tasks.DiscoverPrecompressedAssets.Execute()
   at Microsoft.Build.BackEnd.TaskExecutionHost.Execute()
   at Microsoft.Build.BackEnd.TaskBuilder.<ExecuteInstantiatedTask>d__26.MoveNext()	XXXX	C:\Program Files\dotnet\sdk\9.0.300\Sdks\Microsoft.NET.Sdk.StaticWebAssets\targets\Microsoft.NET.Sdk.StaticWebAssets.Compression.targets	289

**Comment by javiercn** [comment]
@aminmashayekhi your issue seems different. 

Could you file a separate issue and provide a repro project so that we can investigate further?

**Comment by aronbardsley** [comment]
Posting here as a +1 this just happened to me as well. Although my experience is occuring in Azure DevOps build pipelines from seemingly an update which has broken my build pipeline. 

Specifically, seems to have broken the build of one of my test projects:

![Image](https://github.com/user-attachments/assets/0c747c69-8d79-4941-8fae-f10e0d51141d)

**Comment by joshnoe** [comment]
`dotnet clean` clears this up for me.

**Comment by dineshns** [comment]
I am getting same. SDK 9.0.300.

Close VS and then reopen VS, Clean and Build can build once.

**Comment by endizhupani** [comment]
@aronbardsley I had the same issue and adding this task in the pipeline worked in my case. It's the same idea as the proposed fix with global.json, but in the case of DevOps pipelines, adding it to the repo root seems to be ignored for some reason.

```yaml
- task: PowerShell@2
    displayName: 'Create global.json'
    inputs:
      targetType: "inline"
      script: |
        # This is necessary due to an issue when building using the dotnet 9 SDK
        # This will create the global.json within the current workingDir which is needed for publish
        dotnet new globaljson --sdk-version 8.0.0 --roll-forward latestFeature
```

**Comment by figgo** [comment]
It worked for me after I manually removed the files in the "obj" folder

**Comment by jtsom-do** [comment]
I've found that removing the obj and bin folders do help the problem.. until the next time you try and launch the app. Once the bin/obj folders are there, the bug is triggered again.

**Comment by figgo** [comment]
My odj folder is under project root folder not in bin folder. I built
multiple times without issues again

**Comment by dineshns** [comment]
Before that, I tried deleting the obj, .vs, and bin folders. After deleting folders, I could build only once; the next time, I got the same error.

Looks like my issue is fixed after doing this.

- I removed all the .net sdks Used [.net uninstall tool](https://learn.microsoft.com/en-us/dotnet/core/additional-tools/uninstall-tool-overview?pivots=os-windows) - I Had many versions. 3, 6 8 and 9
- Re install .net 9.
- Finally VS2022 repair form VS Installer

Not sure what fixed it. But I can't see the error now.

**Comment by jenkinsonc** [comment]
Having this same problem out of the blue. Nothing in this thread has worked. Solution does not build even on the first opening of Visual Studio.

**Comment by bwinklesky** [comment]
Microsoft, Is there any chance you can make compressing static assets as an "opt-in" feature?  

This has been plaguing builds since November of last year.  I don't need it for my applications, it might be nice to have one day if it ever works appropriately but I would rather develop new features for applications instead of chasing bugs that don't suit me.

**Comment by rhmnmbr** [comment]
try delete the `bin` and `obj` folder and rebuild the project

**Comment by marqdouj** [comment]
Fwiw, I just ran into the same issue on a project that was working fine.

I had accidentally named a razor component the same name as my project. When I tried to build I got the error about the naming issue. I renamed the component and tried to build again; now I started getting the other error.

Tried cleaning, delete bin/obj etc. - nothing worked.
I deleted the code and downloaded the source again from GitHub - works fine now.

So I tried to duplicate this by repeating the original steps above that caused the issue; works/builds fine - no error.

I dunno...

**Comment by JamesFieldist** [comment]
The fundamental issue here that the .net team needs to address is that this "we will not release out of band patches for the SDK" position is dangerous and causing problems.

They did this with Mac last September where they didn't get a patch in quickly enough in April and then Sequoia (15) was released and every Mac user of .NET was SOL for a month if their system upgraded them automatically. Why? Because the .net team screwed up and didn't get a fix in that they knew they needed and then refused to correct the problem with an out of band SDK patch with the fix.

This is the same thing. We have to fight this endlessly for a month because the .net team won't release an out of band SDK patch to fix a major issue. (not quite as bad as "You can't use .net on a mac" but getting there.)

This should be yet another object lesson that this is not a viable position.

**Comment by jtsom-do** [comment]
> try delete the `bin` and `obj` folder and rebuild the project

Unfortunately, as mentioned, this is only a quick stop-gap. Once your project builds successfully, from then on you will get the errors when building. Until you purge the bin and obj folders.

**Comment by lbrpsoftware** [comment]
I'm on VS17.14.6 and dotnet 9.0.301 now.
The bug seems the have been resolved

Edit: VS17.4.7 🤗

**Comment by aminmashayekhi** [comment]
> [@aminmashayekhi](https://github.com/aminmashayekhi) your issue seems different.
> 
> Could you file a separate issue and provide a repro project so that we can investigate further?

Switch the calendar type to Georgian resolved the bug!

**Comment by ruansilva123** [comment]
I'm using Visual Studio 22 and Dotnet 7. I resolved this issue by switching from the current branch to the main branch and rebuilding the API. When I returned to my current branch, the build succeeded.

Maybe the problem was a build cache.🤔

**Comment by javiercn** [comment]
We believe these issues were addressed by #48952 and #49020 that are available in 9.0.301.

The issue was that a cache was not being correctly recomputed, and we were introducing duplicates into the cache that caused issues in later stages of the pipeline.

The original issue was addressed by deduplicating Content items before processing them as part of the pipeline.

If you still experience issues after upgrading to .NET SDK 9.0.302, please file a new issue to let us know.

**Comment by 1944GPW** [comment]
Just encountered this problem when upgrading from .NET 9.0.307 SDK to .NET 10.0.100 under VS2022 Community (64-bit) Version 17.14.20, also removed all obj and bin and cleaned but it's still there,

**Comment by danroth27** [comment]
@1944GPW .NET 10 is only supported with VS 2026.

**Comment by frazambi** [comment]
Still encountering this problem on both VS2026 and Rider 2025.3 on a .NET 10 project.

**Comment by bwinklesky** [comment]
> Still encountering this problem on both VS2026 and Rider 2025.3 on a .NET 10 project.

@frazambi I think they want a new issue to be submitted.  

https://github.com/dotnet/aspnetcore/issues/59625#issuecomment-3146656912

**Comment by 1944GPW** [comment]
> [@1944GPW](https://github.com/1944GPW) .NET 10 is only supported with VS 2026.

That's interesting, I managed to build other solutions upgraded to .NET 10 on VS2022 17.14.20 without any problems. :)
That aside, I installed VS2026, and still have the same issue as I posted above, and like others here.
Also - just reopen this issue please as there seems to be no reason not to.

**Comment by tlevey01** [comment]
Encountering this problem on VS2026 going from .Net 8 to .Net 10.  Mine appears to be saying that blazor.web.js is the duplicate file, but I can't find where it's duplicated and none of the other fixes for the error work.

**Comment by LeeStevens318** [comment]
I was able to reproduce the error when a RCL and Web Project has the same package. 
Here's example: https://github.com/LeeStevens318/csharp-discover-precompressed-assets 

Which worked fine on .net 8, but caused an issue when upgrading to 9/10. 

Microsoft Visual Studio Community
January 2026 Feature Update
Channel:  Stable
Version: 18.2.0

**Comment by tlevey01** [comment]
@LeeStevens318 thank you!  Unfortunately, I had already checked this and in my case, the only shared package is: Microsoft.AspNetCore.App.Internal.Assets.  

In my working .Net 8 version, I have 7 top-level packages shared by the project that won't build in .Net 10 (and over 271 that are Transitive).  After upgrading to .Net 10, I have only one shared package (Microsoft.AspNetCore.App.Internal.Assets, which doesn't exist in my .Net 8 projects) and only 71 Transitive packages.  The main project builds and runs, but this other project fails to build because of the "An item with the same key has already been added" error.

I can't remove Microsoft.AspNetCore.App.Internal.Assets (a note reads: "Implicitly referenced by an SDK. To update the package, update the SDK to which it belongs.").  Any ideas?

**Comment by danroth27** [comment]
@tlevey01 This issue is closed, so if you're still having this problem, please open a new one so that can triage it for investigation.
